spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/source/TestIcebergSourceTablesBase.java

```diff
@@ -737,6 +737,7 @@ public void testHistoryTable() {
     TestHelpers.assertEqualsSafe(historyTable.schema().asStruct(), expected.get(0), actual.get(0));
     TestHelpers.assertEqualsSafe(historyTable.schema().asStruct(), expected.get(1), actual.get(1));
     TestHelpers.assertEqualsSafe(historyTable.schema().asStruct(), expected.get(2), actual.get(2));
+    TestHelpers.assertEqualsSafe(historyTable.schema().asStruct(), expected.get(3), actual.get(3));
```

Comment:
This test was failing, I think due to the problem in #4088. While looking at it, I realized that the last record was not being checked as well.

Reply:
I was wrong about the cause of the history test failure. It turns out that I called the wrong method from `setBranchSnapshot(Snapshot, String)`.